raiden/network/proxies/user_deposit.py

```diff
@@ -101,19 +100,70 @@ def deposit(
         error_prefix = "Call to deposit will fail"
 
         with self.deposit_lock:
-            amount_to_deposit, log_details = self._deposit_preconditions(
-                total_deposit=total_deposit,
-                beneficiary=beneficiary,
-                token=token,
-                block_identifier=block_identifier,
-            )
+            # check preconditions
+            try:
+                previous_total_deposit = self.get_total_deposit(
+                    address=beneficiary, block_identifier=given_block_identifier
+                )
+                current_balance = token.balance_of(
+                    address=self.node_address, block_identifier=given_block_identifier
+                )
+            except (BadFunctionCallOutput, ValueError):
```

Comment:
No no we should not do this. Both exceptions are completely different.

Please follow the pattern used in `set_total_deposit`

Reply:
Ok, changed it to `RaidenUnrecoverableError` as in the other proxy.